core/src/apps/cardano/sign_tx/__init__.py

```diff
@@ -0,0 +1,31 @@
+from trezor import log, messages, wire
+from trezor.enums import CardanoTxSigningMode
+
+from .. import seed
+from .multisig_signer import MultisigSigner
+from .ordinary_signer import OrdinarySigner
+from .plutus_signer import PlutusSigner
+from .pool_owner_signer import PoolOwnerSigner
+
+
+@seed.with_keychain
+async def sign_tx(
+    ctx: wire.Context, msg: messages.CardanoSignTxInit, keychain: seed.Keychain
+) -> messages.CardanoSignTxFinished:
+    signer_type = {
```

Comment:
I'd consider splitting the dict declaration just so typing would be a bit better:
```suggestion
    signer_types: dict[CardanoTxSigningMode, Type[Signer]] = {
        CardanoTxSigningMode.ORDINARY_TRANSACTION: OrdinarySigner,
        CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER: PoolOwnerSigner,
        CardanoTxSigningMode.MULTISIG_TRANSACTION: MultisigSigner,
        CardanoTxSigningMode.PLUTUS_TRANSACTION: PlutusSigner,
    }
    signer_type = signer_types[msg.signing_mode]
```